pkg/virt-config/virt-config.go

```diff
@@ -209,11 +209,31 @@ func (c *ClusterConfig) GetPermittedHostDevices() *v1.PermittedHostDevices {
 	return c.GetConfig().PermittedHostDevices
 }
 
-func (c *ClusterConfig) GetDesiredMDEVTypes(nodeName string) []string {
+func (c *ClusterConfig) GetDesiredMDEVTypes(node *k8sv1.Node) []string {
 	mdevTypesConf := c.GetConfig().MediatedDevicesConfiguration
 	if mdevTypesConf == nil {
 		return []string{}
 	}
+	nodeMdevConf := mdevTypesConf.MediatedDevicesTypes.NodeMediatedDeviceTypesConfig
+	if nodeMdevConf != nil {
+		mdevTypesMap := make(map[string]struct{})
+		for _, nodeConfig := range nodeMdevConf {
+			for key, _ := range nodeConfig.NodeSelector {
+				if _, exist := node.Labels[key]; exist {
```

Comment:
maybe all keys need to match the node? not sure. we'd want to make sure the matching behavior is the same that the k8 sscheduler uses

Reply:
Yeah. I wasn't sure what would we want to do here. 

For example, what happens if the node matches more than one node selector?
Right now, the code just merges the types from all matching selectors. 

In the case of the following config and when the node has both label1 and label2:
```
mediatedDevicesConfiguration:
      mediatedDevicesTypes:
        - nvidia-222
        - nvidia-228
        - i915-GVTg_V5_4
      nodeMediatedDevices:
      - nodeSelector:
        label1: ""
        mediatedDevicesTypes:
        - nvidia-223
      - nodeSelector:
        label2: ""
        mediatedDevicesTypes:
        - nvidia-229
```

I think i'd expect to configure  nvidia-223 and nvidia-223 on this node.

What do you think?

Reply:
i'd expect the merger like you outline.